src/Stratis.Bitcoin.Features.SmartContracts/Rules/SmartContractCoinViewRuleLogic.cs

```diff
@@ -76,7 +77,12 @@ public async Task RunAsync(Func<RuleContext, Task> baseRunAsync, RuleContext con
             this.logger.LogDebug("Block to validate '{0}'", block.GetHash());
 
             // Get a IStateRepositoryRoot we can alter without affecting the injected one which is used elsewhere.
-            uint256 blockRoot = ((ISmartContractBlockHeader)context.ValidationContext.ChainedHeaderToValidate.Previous.Header).HashStateRoot;
+            BlockHeader prevHeader = context.ValidationContext.ChainedHeaderToValidate.Previous.Header;
+            uint256 blockRoot;
+            if (!(prevHeader is PosBlockHeader posHeader) || posHeader.HasSmartContractFields)
+                blockRoot = ((ISmartContractBlockHeader)prevHeader).HashStateRoot;
+            else
+                blockRoot = new uint256("21B463E3B52F6201C0AD6C991BE0485B6EF8C092E64583FFA655CC1B171FE856");
```

Comment:
please move this sting to a variable and comment it

Reply:
Done.